server.go

```diff
@@ -243,21 +248,73 @@ func newServer(listenAddrs []string, chanDB *channeldb.DB, cc *chainControl,
 	}
 	s.htlcSwitch = htlcSwitch
 
+	// If enabled, use either UPnP or NAT-PMP to automatically configure
+	// port forwarding for users behind a NAT.
+	switch {
+	case cfg.UPnP:
+		srvrLog.Info("Scanning local network for a UPnP enabled device")
+
+		ctx, cancel := context.WithTimeout(
+			context.Background(), 30*time.Second,
+		)
+		defer cancel()
+		upnp, err := nat.DiscoverUPnP(ctx)
+		if err != nil {
+			srvrLog.Errorf("Unable to discover a UPnP enabled "+
+				"device on the local network: %v", err)
+			break
+		}
+
+		s.natTraversal = upnp
+	case cfg.NATPMP:
+		srvrLog.Info("Scanning local network for a NAT-PMP enabled device")
+
+		pmp, err := nat.DiscoverPMP(30 * time.Second)
+		if err != nil {
+			srvrLog.Errorf("Unable to discover a NAT-PMP enabled "+
+				"device on the local network: %v", err)
+			break
+		}
+
+		s.natTraversal = pmp
+	}
+
+	// If we were requested to automatically configure port forwarding,
+	// we'll use the ports that the server will be listening on.
+	if s.natTraversal != nil {
+		listenPorts := make([]uint16, 0, len(listenAddrs))
+		for _, listenAddr := range listenAddrs {
```

Comment:
I suspect most people using this want to let a port be chosen automatically. 

(guilty 😂)